src/coreclr/vm/excep.cpp

```diff
@@ -5222,21 +5205,16 @@ static LONG ThreadBaseExceptionFilter_Worker(PEXCEPTION_POINTERS pExceptionInfo,
 
     LOG((LF_EH, LL_INFO100, "ThreadBaseExceptionFilter_Worker: Enter\n"));
 
-    ThreadBaseExceptionFilterParam *pParam = (ThreadBaseExceptionFilterParam *) pvParam;
-    UnhandledExceptionLocation location = pParam->location;
-
     Thread* pThread = GetThread();
 
 #ifdef _DEBUG
     if (CLRConfig::GetConfigValue(CLRConfig::INTERNAL_BreakOnUncaughtException) &&
-        !(swallowing && ExceptionIsAlwaysSwallowed(pExceptionInfo)) &&
-        !(location == ClassInitUnhandledException && pThread->IsRudeAbortInitiated()))
+        !(swallowing && ExceptionIsAlwaysSwallowed(pExceptionInfo)))
         _ASSERTE(!"BreakOnUnCaughtException");
 #endif
 
-    BOOL doDefault =  ((location != ClassInitUnhandledException) &&
-                       (pExceptionInfo->ExceptionRecord->ExceptionCode != STATUS_BREAKPOINT) &&
-                       (pExceptionInfo->ExceptionRecord->ExceptionCode != STATUS_SINGLE_STEP));
+    BOOL doDefault = (pExceptionInfo->ExceptionRecord->ExceptionCode != STATUS_BREAKPOINT) &&
+                     (pExceptionInfo->ExceptionRecord->ExceptionCode != STATUS_SINGLE_STEP);
 
     if (swallowing)
```

Comment:
This method was always called with `swallowing == TRUE` so the rest of the method after this ifdef is dead code.